src/validators/required.ts

```diff
@@ -0,0 +1,52 @@
+import { FieldValidationFunctionSync, ValidationResult } from '../model';
+
+let errorMessage = 'Please fill in this mandatory field.';
+
+export const VALIDATOR_TYPE = 'REQUIRED';
+
+export const setErrorMessage = (message: string) => {
+  errorMessage = message;
+};
+
+export interface RequiredArgs {
+  trim: boolean;
+}
+
+const DEFAULT_PARAMS: RequiredArgs = { trim: true };
+
+const createDefaultValidationResult = (): ValidationResult => ({
+  succeeded: true,
+  type: VALIDATOR_TYPE,
+  message: '',
+});
+
+const isStringValid = (value: string, trim: boolean): boolean => {
+  return trim ? value.trim().length > 0 : value.length > 0;
+};
+
+const isValidField = (value: any, trim: boolean): boolean => {
+  return typeof value === 'string'
+    ? isStringValid(value, trim)
+    : value === true || typeof value === 'number';
+};
+
+export const required: FieldValidationFunctionSync = (
+  value,
```

Comment:
I think could be a goodie if we provide an object instead of 4 parameters. What do you think?

```
// model.ts
interface ValidatorArgs {
  value: any,
  values?: any,
  customArgs?: any,
  message?: string | string[]
}

//single-field-dispatcher.ts
const fireValidation = (
  value,
  values,
  fieldValidation: FieldValidation) => {
  const validationFn = getValidationFn(fieldValidation);
  const validationArgs = {
    value,
    values,
    customArgs: fieldValidation['customArgs'],
    message: fieldValidation['message']
}
  return validationFn(validationArgs).then(checkValidationResult);
}


// required.ts

export const required: FieldValidationFunctionSync = (validatorArgs: ValidatorArgs) => {
  const { value, customArgs, customMessage } = validatorArgs;
}
```

Reply:
If we create a structure it could help mitigating the issue of customArgs / message optional parameters order.

On the other hand this will be exposed to users? not sure if it can be the best api